docs/apps/components/data-grid.md

```diff
@@ -0,0 +1,208 @@
+# Data Grid
+
+## Description
+
+**Data Grid** is a tabular data display component. It enables you to add different functionalities.
+
+<img src= "/apps/components/img/datagrid.png">
+
+## Use Case
+
+1. Displaying and Editing Tabular Data.
+2. Collecting and Managing Lists.
+3. Building Dynamic Forms.
+4. Integrating with External Data.
+
+### Interactivity
+
+1. Allow users to directly edit data within cells, eliminating the need for separate edit forms or modals.
+2. Provide tools for users to sort data by different columns and apply filters to narrow down the displayed information.
+3. Pagination
+4. Expandable Rows
+5. Enforce data integrity by implementing validation rules to ensure data accuracy and consistency.
+
+## API
+
+### Events
+
+| **Name**| **Description**|
+|---------|----------------|
+| `onCellClicked`| The onCellClicked event occurs when the cell in a data grid is clicked|
+| `onLinkClicked`| The onLinkClicked event occurs when the link is clicked|
+
+### Methods
+
+| **Name**| **Description**|**Parameters**|
+|---------|----------------|--------------|
+|`this.addClassName`|Add style class to a form item|`(fields: String \| String[], className: String)`|
+|`this.removeClassName`|Remove style class from a form item|`(fields: String \| String[], className: String)`|
+|`this.getComponent`|Returns a component whose id has been passed as a parameter|`(component_ID: String : Object)`|
```

Comment:
Please remember that the return value of a function is shown outside the parameter brackets. So in this case, 
the parameters section would have to be as follows: `(component_Id: String) : Object`

Reply:
done

Reply:
done